doc/feature/cppPlugins.md

```diff
@@ -0,0 +1,132 @@
+C++ Plugin API for CPU compressors  {#data}
+============
+
+This document specifies a modern API for the in-memory C plugin API for
+lossless byte data compressors. It will not replace the 2D image
+compressors and CPU-GPU transfer plugins.
+
+## Requirements
+
+* Support all features used by current lossless byte compressors:
+    * Lossless compression of binary blobs
+    * Unified API for all compression engines
+    * Other components can add compression engines
+* Information provided for each compressor:
+    * Compression speed (relative to RLE compressor)
+    * Compression ratio (relative to uncompressed data)
+* Compressors may produce multiple output chunks
+* Compressor engines are instantiated to allow reuse of internal state
+  and memory allocations
+* Easy to use API for plugin implementers
+
+## Dependency Changes
+
+* New: Using Lunchbox plugin API for loading plugins
+
+## API
+
+    class CompressorInfo
+    {
+    public:
+        std::string name;
+        float ratio; //!< Normalized 0..1 size after compression
+        float speed; //!< Relative speed compared to RLE compressor
+
+        typedef std::function< Compressor* ()> Constructor;
+        Constructor create;
+    };
+    typedef std::vector< CompressorInfo > CompressorInfos;
+
+    class Registry
+    {
+    public:
+        static Registry& getInstance();
+        template< class P > bool registerEngine( CompressorInfo info )
+
+        const CompressorInfos& getInfos() const;
+        CompressorInfo choose();
+        CompressorInfo find( const std::string& name );
+    };
+
+    class Compressor
+    {
+    public:
+        typedef lunchbox::Bufferb Result;
+        typedef std::vector< Result > Results;
+
+        virtual const Results& compress( const uint8_t* data, size_t size ) = 0;
```

Comment:
A compressor may return 1..n compressed chunks (of getChunkSize() uncompressed input data) which is restriped by the Slicer into m (m<=n) block for storage in memcached.